packages/ODEs/src/app.ts

```diff
@@ -931,4 +946,69 @@ export class DiffStudio {
         ${(err instanceof Error) ? err.message : ERROR_MSG.SCRIPTING_ISSUE}`);
     }
   }
+
+  /** Try to solve IVP */
+  private async tryToSolve(ivp: IVP): Promise<void> {
+    const optionsBuf = ivp.solverSettings;
+    ivp.solverSettings = DEFAULT_OPTIONS.SCRIPTING;
+
+    try {
+      const scriptText = getScriptLines(ivp, true, true).join('\n');
+      ivp.solverSettings = optionsBuf;
+      const script = DG.Script.create(scriptText);
+
+      // try to call computations - correctness check
+      const params = getScriptParams(ivp);
+      const call = script.prepare(params);
+      await call.call();
+    } catch (err) {
+      if (!(err instanceof CallbackAction))
+        throw new Error((err instanceof Error) ? err.message : ERROR_MSG.SCRIPTING_ISSUE);
+    }
+  }
+
+  /** Close previousely opened performance dialog */
+  private closePerformanceDlg(): void {
+    if (this.performanceDlg) {
+      this.performanceDlg.close();
+      this.performanceDlg = null;
+    }
+  }
+
+  /** Show performance dialog */
+  private async showPerformanceDlg(ivp: IVP, inputsPath: string): Promise<void> {
```

Comment:
I don't see any async code inside, I think this method should be sync.
Also, it might probably be a good idea for it to return the dialog (to be able to control the result later)

Reply:
Corrected